packages/components/src/custom-select-control-v2/types.ts

```diff
@@ -0,0 +1,58 @@
+/**
+ * External dependencies
+ */
+// eslint-disable-next-line no-restricted-imports
+import type * as Ariakit from '@ariakit/react';
+
+export type CustomSelectContext =
+	| {
+			/**
+			 * The store object returned by Ariakit's `useSelectStore` hook.
+			 */
+			store: Ariakit.SelectStore;
+	  }
+	| undefined;
+
+export type CustomSelectProps = {
+	/**
+	 * The child elements. This should be composed of CustomSelect.Item components.
+	 */
+	children: React.ReactNode;
+	/**
+	 * An optional default value for the control. If left `undefined`, the first
+	 * non-disabled item will be used.
+	 */
+	defaultValue?: string | string[];
+	/**
+	 * Label for the control.
+	 */
+	label?: string;
+	/**
+	 * A function that receives the new value of the input.
+	 */
+	onChange?: ( newValue: string | string[] ) => void;
+	/**
+	 * Can be used to render select UI with custom styled values.
+	 */
+	styledValue?: ( selectValue: string | string[] ) => React.ReactNode;
+	/**
+	 * The size of the control.
+	 */
+	size?: 'default' | 'large';
+	/**
+	 * Can be used to externally control the value of the control.
+	 */
+	value?: string | string[];
```

Comment:
Assuming that we're not going to support multiple selection, at least for this first iteration of the component, then it's probably better to have `value`, `defaultValue` and `onChange` to just deal with `string` (and therefore, remove `string[]`).